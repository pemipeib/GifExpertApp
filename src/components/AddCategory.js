import React, {useState} from 'react';
import PropTypes from 'prop-types';


const AddCategory = ({ setCategories }) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    
    const handleSubmit = (e) => {
        e.preventDefault(); //para que no refresque la pÃ¡gina
        if (inputValue.trim().length > 2){
            setCategories(cats => [ inputValue, ...cats,]);
            setInputValue('');
        }
    }
    
    return (
        <form onSubmit={ handleSubmit }>
            {/*Este parrafo esta solo para las pruebas*/}
            <p>{inputValue}</p>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
                placeholder={'Nombre de la busqueda'}
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
