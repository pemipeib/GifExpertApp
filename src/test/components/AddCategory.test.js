import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from "enzyme";

import AddCategory from "../../components/AddCategory";

describe('Pruebas en AddCategory', () =>{
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={setCategories}/>);
    
    beforeEach( () =>{
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={setCategories}/>);
    });
    
    test('Debe mostrar AddCategory correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {target: {value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
    });
    
    test('No debe de postear el sumbit', () => {
        wrapper.find('form').simulate('submit', {preventDefault (){}}) //Forma corta del preventDefault en este caso
        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Hola mundo';
        wrapper.find('input').simulate('change', {target: {value}});
        wrapper.find('form').simulate('submit', {preventDefault (){}}) //Forma corta del preventDefault en este caso
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});
