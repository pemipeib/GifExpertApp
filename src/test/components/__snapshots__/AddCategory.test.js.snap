// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory Debe mostrar AddCategory correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Nombre de la busqueda"
    type="text"
    value=""
  />
</form>
`;
